src/py21cmmc/core.py

```diff
@@ -645,3 +645,245 @@ def convert_model_to_mock(self, ctx):
+
+    Parameters
+    ----------
+    z_extrap_min : minimal z for reionization in CLASS. should basically always be set to 0.
+    z_extrap_max : maximal z for reionization in CLASS. depends on the reionization model.
+    z_HeI : redshift of the first helium reionization. CLASS models helium reionzation with a tanh centered around zHeI.
+    z_HeII : redshift of the second helium reionization. CLASS models helium reionzation with a tanh centered around zHeII.
+    use_21cmfast : whether or not using EoR history from 21cmfast.
+    """
+
+    def __init__(
+        self,
+        verbose=0,
+        z_extrap_min=0,
+        z_extrap_max=20,
+        z_HeI=4,
+        z_HeII=3,
+        use_21cmfast=True,
+        user_params=None,
+        flag_options=None,
+        astro_params=None,
+        cosmo_params=None,
+        regenerate=True,
+        change_seed_every_iter=False,
+        ctx_variables=("brightness_temp", "xH_box"),
+        initial_conditions_seed=None,
+        global_params=None,
+        **io_options,
+    ):
+
+        super().__init__(io_options.get("store", None))
+
+        if not use_21cmfast:
+            self.user_params = p21.UserParams(user_params)
+            self.flag_options = p21.FlagOptions(flag_options)
+            self.astro_params = p21.AstroParams(astro_params)
+            self.cosmo_params = p21.CosmoParams(cosmo_params)
+            self.change_seed_every_iter = change_seed_every_iter
+            self.initial_conditions_seed = initial_conditions_seed
+
+            self.regenerate = regenerate
+            self.ctx_variables = ctx_variables
+
+            self.global_params = global_params or {}
+
+            self.io_options = {
+                "store": {},  # (derived) quantities to store in the MCMC chain.
+                "cache_dir": None,  # where full data sets will be written/read from.
+                "cache_mcmc": False,  # whether to cache ionization data sets
+                # (done before parameter retention step)
+            }
+
+            self.io_options.update(io_options)
+
+        try:
+            from classy import Class
+
+            if verbose > 0:
+                print("import CLASS")
+            global cosmo
+            cosmo = Class()
+            self.verbose = verbose
+            self.z_extrap_min = z_extrap_min
+            self.z_extrap_max = z_extrap_max
+            self.z_HeI = z_HeI
+            self.z_HeII = z_HeII
+            self.use_21cmfast = use_21cmfast
+        except AttributeError:
+            raise AttributeError(
+                "You must have compiled the classy.pyx file. Please go to "
+                + "/path/to/class/python and run the command\n "
+                + "python setup.py build"
+            )
+
+    def setup(self):
+        """Perform any post-init setup of the object."""
+        CoreBase.setup(self)
```

Comment:
It's better to do `super().setup()` so if we ever change the name of `CoreBase` it will still work.